### ISSUE TITLE
Track open time

### ISSUE THREAD

**Issue by DanBrooker** [jnunemaker/resilient#14]
Is there something internally that I could use to track how long the circuit has been open?
I'm just this is wrap a third party API and would like to get a notification if it's been down for say 5 mins.

Also thanks for open sourcing this. 👍

**Comment by DanBrooker** [comment]
I logged some circuit breakers and the metrics contain a timestamp_start and a timestamp_end which I think could be used. 

I definitely could measure the time myself, but am hoping I can use the circuit breakers existing data.

**Comment by jnunemaker** [comment]
Missed this somehow. Sorry! I don't think there is a great way to do so currently. At GitHub, we publish stats to graphite and have nagios checks that monitor success/failure data in graphite so we haven't needed an event for open/close. I think if an open/close event were needed it would go in these methods: https://github.com/jnunemaker/resilient/blob/95d9067230de475a9429f6d0ac9c2ac7c6f4a992/lib/resilient/circuit_breaker.rb#L108-L117. Any interesting in working up a PR to look at?

**Comment by DanBrooker** [comment]
I can try add some ActiveSupport::Notifications (`instrument`) if that's what you mean

**Comment by jnunemaker** [comment]
Yeah, I think you would just need something along the lines of [this](https://github.com/jnunemaker/resilient/blob/95d9067230de475a9429f6d0ac9c2ac7c6f4a992/lib/resilient/circuit_breaker.rb#L128).

**Comment by jnunemaker** [comment]
All done and released in 0.5.1.
